### ISSUE TITLE
Upgrade bouncycastle to org.bouncycastle:bcprov-jdk15to18:1.70

### ISSUE THREAD

**Issue by sebastienlabine** [thebergamo/react-native-fbsdk-next#323]
As expo dropped their sdk for facebook we must rely on this native module. However, it's impossible to build any app using expo and this sdk as the [org.bouncycastle](bouncycastle:bcprov targets a different version.

I found a workaround here https://github.com/stripe/stripe-android/issues/3173

app/build.gradle
```

{
configurations.all {
    c -> c.resolutionStrategy.eachDependency {
        DependencyResolveDetails dependency ->
            println dependency.requested.group
             if (dependency.requested.group == 'org.bouncycastle') {
                dependency.useTarget 'org.bouncycastle:bcprov-jdk15to18:1.68'
            }
    }
}
```

Could it be possible to upgrade the dependency to org.bouncycastle:bcprov-jdk15to18:1.70

**Comment by mikehardy** [comment]
Hi there! I ran `./gradlew :app:dependencies > dependencies.txt` from the example here, and the problem is the gamingservices component has incorrectly specified their test dependencies as non-test dependencies, so robolectric is shipping as a transitive dependency of the upstream SDK, and it has the transitive dependency you mention.

https://github.com/facebook/facebook-android-sdk/blob/dfbe44cc2fe796dd007447922c6e15fd418d56b5/facebook-gamingservices/build.gradle.kts#L37

This is out of our control as we just include them, and the error is upstream. Fastest resolution will be for you to file an issue in their repo using this information, referencing this one. When they do a new release with a fix, you'll get it automatically in your next build because our android dependency version range is set to `15.+`, allowing semver minor and patch to come in immediately in project builds

**Comment by Wituz** [comment]
I've written a very bare config plugin for those using Expo, managed or semi-managed workflow, as a temporary workaround. 

Put this in `plugins/withFixedBouncyCastleAndroidVersion.js`:
```javascript
const { createRunOncePlugin, withAppBuildGradle } = require('@expo/config-plugins');

const withFixedBouncyCastleAndroidVersion = (config) => {
  return withAppBuildGradle(config, config => {
    config.modResults.contents = setBouncyCastleVersion(config.modResults.contents)
    return config;
  });
}

const setBouncyCastleVersion = (buildGradle) => {

    const stringToInsert = `
    // Manually fix bouncycastle version until facebook-android-sdk is updated
    configurations.all {
      c -> c.resolutionStrategy.eachDependency {
          DependencyResolveDetails dependency ->
              println dependency.requested.group
              if (dependency.requested.group == 'org.bouncycastle') {
                  dependency.useTarget 'org.bouncycastle:bcprov-jdk15to18:1.70'
              }
      }
    }\n`
    
    const insertBelowLine = "android {"
    if (!buildGradle.includes(stringToInsert)) {
      buildGradle = buildGradle.replace(insertBelowLine, insertBelowLine + stringToInsert)
    }

    return buildGradle
}

module.exports = createRunOncePlugin(
  withFixedBouncyCastleAndroidVersion,
  'withFixedBouncyCastleAndroidVersion',
  '1.0.0'
);
```

Then in app.json or in app.config.js, apply the plugin
```json
"plugins": ["./plugins/withFixedBouncyCastleAndroidVersion"]
```

And your app should be able to build

**Comment by sebastienlabine** [comment]
@Wituz I fixed your code to allow multiple expo prebuild runs without having duplicate code:

```ts
const { createRunOncePlugin, withAppBuildGradle } = require('@expo/config-plugins');
const { mergeContents } = require('@expo/config-plugins/build/utils/generateCode');

const withFixedBouncyCastleAndroidVersion = (config) => {
    return withAppBuildGradle(config, config => {
        config.modResults.contents = setBouncyCastleVersion(config.modResults.contents)
        return config;
    });
}

const TAG = "@smartbills/expo-bouncy-castle-fix"

const setBouncyCastleVersion = (src) => {

    const configurationFix = `
    configurations.all {
      c -> c.resolutionStrategy.eachDependency {
          DependencyResolveDetails dependency ->
              println dependency.requested.group
              if (dependency.requested.group == 'org.bouncycastle') {
                  dependency.useTarget 'org.bouncycastle:bcprov-jdk15to18:1.70'
              }
      }
    }`

    var mergeResults = mergeContents({
        tag: TAG,
        src: src,
        newSrc: configurationFix,
        anchor: /android {/gm,
        offset: 1,
        comment: `// `
    });

    return mergeResults.contents;
}

module.exports = createRunOncePlugin(
    withFixedBouncyCastleAndroidVersion,
    'withFixedBouncyCastleAndroidVersion',
    '1.0.0'
);
```

**Comment by mikehardy** [comment]
There is an in-progress PR over at #327 to the Expo plugin - but it has an error in it. Looks like some good config-plugin code here - if someone wanted to take the commit(s) from 327 and work in the stuff here and post a PR I'd be happy to merge it

**Comment by sebastienlabine** [comment]
@mikehardy Upstream issue was fixed in their 15.0.2 release, meaning this should be fixed here as well?

**Comment by mikehardy** [comment]
Indeed! Saw the release a couple hours ago. This will autofix for people next time they build android
